src/dsql/dsql.cpp

```diff
@@ -439,6 +441,23 @@ static dsql_dbb* init(thread_db* tdbb, Jrd::Attachment* attachment)
 	return attachment->att_dsql_instance;
 }
 
+// Use SEH frame when preparing user requests to catch possible stack overflows
+static DsqlRequest* safePrepareRequest(thread_db* tdbb, dsql_dbb* database, jrd_tra* transaction,
+	ULONG textLength, const TEXT* text, USHORT clientDialect, bool isInternalRequest)
+{
+	if (isInternalRequest)
+		return prepareRequest(tdbb, database, transaction, textLength, text, clientDialect, isInternalRequest);
```

Comment:
`isInternalRequest -> true` maybe ? And `isInternalRequest -> false` below.

Reply:
No objection, just curious - for what ? ;)

Reply:
Two absolutely identical calls looks suspicious at the first glance ;-) You already checked for `isInternalRequest`, so the result is known. But I don't insist, it's a highly subjective point.

Reply:
Ok, done